userdocs/diagnostics/multiple-class-inheritance.md

```diff
@@ -0,0 +1,37 @@
+# Multiple Inheritance of Classes
+
+In some programming languages, a class can have multiple superclasses. This feature is called multiple inheritance and is unsupported in Swift. However, using protocols, one can achieve a similar effect as multiple inheritance.
+
+Consider the following example with a class `Bird` conforming to two protocols `Animal` and `Flyable`.
+
+```swift
+protocol Flyable {
+    func fly()
+}
+
+extension Animal {
+    func move() {
+        print("Moving...")
+    }
+}
+
+protocol Animal {
+    func move()
+}
+
+extension Flyable {
+    func fly() {
+        print("Flying...")
+    }
+}
+
+class Bird: Animal, Flyable {
+    // move() and fly() use default implementations
+}
+```
+
+Since `Animal` and `Flyable` provide default implementations for their protocol requirements, `Bird` can conform to both without implementing `fly()` or `move()` and the default implementation will be used. This effect is similar to class inheritance where a subclass inherits properties, subscripts and methods from a superclass. The default implementation can be overriden by defining `fly()` or `move()` without an `override` keyword.
```

Comment:
```suggestion
```

As noted previously, I do not think this educational note needs to explain or demonstrate how default implementations work. It's pretty inherent to basic usage of protocols; if a user doesn't know about protocols, they can follow the links given below to read more in _TSPL_ which explain it adequately.